src/title-card/titleCardEditForm.ts

```diff
@@ -0,0 +1,343 @@
+/* eslint-disable no-underscore-dangle */
+import { css, html, LitElement, nothing } from 'lit';
+import { customElement, property, query, state } from 'lit/decorators.js';
+import { HomeAssistant, HuiElementEditor } from '../types';
+import { TitleCardConfig, TitleCardEditFormParams } from './showTitleCardEditForm';
+import { ifDefined } from 'lit/directives/if-defined.js';
+import { haStyleDialog, haStyleDialogFixedTop } from '../helpers/ha-dialog-styles';
+
+export const CUSTOM_TYPE_PREFIX = 'custom:';
+
+const isCustomType = (type: string) =>
+    type.startsWith(CUSTOM_TYPE_PREFIX);
+
+const getCustomCardEntry = (type: string) =>
+    window.customCards?.find((card) => card.type === type); // NOSONAR es2019
+
+const stripCustomPrefix = (type: string) =>
+    type.replace(CUSTOM_TYPE_PREFIX, '');
+
+@customElement('expander-card-title-card-edit-form')
+export class TitleCardEditForm extends LitElement {
+    @property({ attribute: false }) public hass?: HomeAssistant;
+    @property({ type: Boolean, reflect: true }) public large = false;
+    @property({ attribute: false }) public lovelace?: unknown;
+
+    @state() private _params?: TitleCardEditFormParams;
+    @state() private _config: TitleCardConfig = {};
+    @state() private _cardGUIMode = true;
+    @state() private _cardGUIModeAvailable = true;
+    @state() private _error = false;
+
+    @query('hui-card-element-editor') private _cardEditorEl?: HuiElementEditor; // NOSONAR Lit @query decorator updates
```

Comment:
@MelleD SONAR wants this to be readonly. The Lit query decorator calls querySelector, so its not readonly but will be updated by Lit. Perhaps SONAR has some Lit defaults to include?